code/game/machinery/slotmachine.dm

```diff
@@ -275,12 +303,15 @@
 
 /obj/machinery/computer/slot_machine/proc/give_money(amount)
 	var/amount_to_give = money >= amount ? amount : money
-	var/surplus = amount_to_give - give_coins(amount_to_give)
+	var/surplus = amount_to_give - give_payout(amount_to_give)
 	money = max(0, money - amount)
 	balance += surplus
 
-/obj/machinery/computer/slot_machine/proc/give_coins(amount)
-	var/cointype = obj_flags & EMAGGED ? /obj/item/coin/iron : /obj/item/coin/silver
+/obj/machinery/computer/slot_machine/proc/give_payout(amount)
+	if(paymode == HOLOCHIP)
+		var/cointype = /obj/item/holochip
+	else
+		var/cointype = obj_flags & EMAGGED ? /obj/item/coin/iron : /obj/item/coin/silver
```

Comment:
Should be initialized here, not defined as cointype stops existing past this condition due to it.
Should be
`cointype = obj_flags & EMAGGED ? /obj/item/coin/iron : /obj/item/coin/silver`